Multiple TenantControlTest failures [HZ-1843]

- Possible commit `Adds tenant control (commit: 02b2b45) (detail)`
- Fails on `Hazelcast-4.master-Solaris-OracleJDK8`
- Fails on [Build #255 (Dec 19, 2020 7:20:46 PM)](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4.master-Solaris-OracleJDK8/com.hazelcast$hazelcast/255/#showFailuresLink)
- Sample Error
```
expected:<1> but was:<0>
```
- Sample Stacktrace
```
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.spi.tenantcontrol.TenantControlTest.basicSetTest(TenantControlTest.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```
- Kindly check

It looks also need update on code-samples repo for tenant updates. ⚠️ 
```
00:03:44.608 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project tenant-control: Compilation failure: Compilation failure: 
00:03:44.608 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-4.master-code-samples-OracleJDK8/hazelcast-code-samples/spi/tenant-control/src/main/java/multitenant/MultiTenantControl.java:[9,8] multitenant.MultiTenantControl is not abstract and does not override abstract method clearThreadContext() in com.hazelcast.spi.tenantcontrol.TenantControl
00:03:44.608 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-4.master-code-samples-OracleJDK8/hazelcast-code-samples/spi/tenant-control/src/main/java/multitenant/MultiTenantControl.java:[17,5] method does not override or implement a method from a supertype
00:03:44.608 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-4.master-code-samples-OracleJDK8/hazelcast-code-samples/spi/tenant-control/src/main/java/multitenant/MultiTenantControl.java:[25,16] incompatible types: multitenant.TenantCloseable cannot be converted to com.hazelcast.spi.tenantcontrol.TenantControl.Closeable
00:03:44.608 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-4.master-code-samples-OracleJDK8/hazelcast-code-samples/spi/tenant-control/src/main/java/multitenant/MultiTenantControl.java:[28,5] method does not override or implement a method from a supertype
00:03:44.608 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-4.master-code-samples-OracleJDK8/hazelcast-code-samples/spi/tenant-control/src/main/java/multitenant/MultiTenantControlFactory.java:[7,8] multitenant.MultiTenantControlFactory is not abstract and does not override abstract method isClassesAlwaysAvailable() in com.hazelcast.spi.tenantcontrol.TenantControlFactory
00:03:44.608 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-4.master-code-samples-OracleJDK8/hazelcast-code-samples/spi/tenant-control/src/main/java/multitenant/MultiTenantControlFactory.java:[9,5] method does not override or implement a method from a supertype
00:03:44.608 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-4.master-code-samples-OracleJDK8/hazelcast-code-samples/spi/tenant-control/src/main/java/multitenant/MultiTenantControlFactory.java:[11,33] cannot find symbol
```

--- a/hazelcast/src/test/java/com/hazelcast/spi/tenantcontrol/TenantControlTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/tenantcontrol/TenantControlTest.java
@@ -143,8 +143,8 @@ public class TenantControlTest extends TenantControlTestSupport {
         assertEquals(4, setTenantCount.get());
         assertEquals(4, closeTenantCount.get());
         assertEquals(1, registerTenantCount.get());
-        assertEquals(1, unregisterTenantCount.get());
-        assertEquals(3, clearedThreadInfoCount.get());
+        assertEqualsEventually(1, unregisterTenantCount);
+        assertEqualsEventually(3, clearedThreadInfoCount);
     }
 
     @Test
@@ -177,7 +177,7 @@ public class TenantControlTest extends TenantControlTestSupport {
         // operation does not need tenant context
         assertEquals(1, setTenantCount.get());
         assertEquals(1, registerTenantCount.get());
-        assertEquals(1, unregisterTenantCount.get());
+        assertEqualsEventually(1, unregisterTenantCount);
     }
 
     @Test
@@ -194,7 +194,7 @@ public class TenantControlTest extends TenantControlTestSupport {
         // set tenant is called twice for add, once for remove and clear
         assertEquals(4, setTenantCount.get());
         assertEquals(1, registerTenantCount.get());
-        assertEquals(1, unregisterTenantCount.get());
+        assertEqualsEventually(1, unregisterTenantCount);
     }
 
     @Test
@@ -211,7 +211,7 @@ public class TenantControlTest extends TenantControlTestSupport {
         // set tenant is called twice for add, once for remove and clear
         assertEquals(4, setTenantCount.get());
         assertEquals(1, registerTenantCount.get());
-        assertEquals(1, unregisterTenantCount.get());
+        assertEqualsEventually(1, unregisterTenantCount);
     }
 
     @Test
@@ -228,7 +228,7 @@ public class TenantControlTest extends TenantControlTestSupport {
         // set tenant is called twice for add, once for remove and clear
         assertEquals(4, setTenantCount.get());
         assertEquals(1, registerTenantCount.get());
-        assertEquals(1, unregisterTenantCount.get());
+        assertEqualsEventually(1, unregisterTenantCount);
     }
 
     private void assertCacheTenantControlCreated(HazelcastInstance instance) {

